dbms/src/Storages/DeltaMerge/tests/gtest_dm_segment.cpp

```diff
@@ -1024,6 +1025,112 @@ try
 }
 CATCH
 
+TEST_F(SegmentTest, ColumnFileBigRangeGreaterThanSegment)
+try
+{
+    auto write_column_file_big = [&](size_t begin, size_t end, size_t block_num) {
+        WriteBatches ingest_wbs(*dm_context->storage_pool, dm_context->getWriteLimiter());
+        auto delegator = storage_path_pool->getStableDiskDelegator();
+        auto parent_path = delegator.choosePath();
+        auto file_id = storage_pool->newDataPageIdForDTFile(delegator, __PRETTY_FUNCTION__);
+
+        std::vector<BlockInputStreamPtr> streams;
+        streams.reserve(block_num);
+        size_t step = (end - begin) / block_num;
+        for (size_t i = 0; i < block_num; ++i)
+        {
+            Block block = DMTestEnv::prepareSimpleWriteBlock(begin + i * step, begin + (i + 1) * step, false);
+            auto stream = std::make_shared<OneBlockInputStream>(block);
+            streams.push_back(stream);
+        }
+        auto input_stream = std::make_shared<ConcatBlockInputStream>(std::move(streams), dmContext().tracing_id);
+
+        auto dm_file = writeIntoNewDMFile(*dm_context, table_columns, input_stream, file_id, parent_path);
+        ingest_wbs.data.putExternal(file_id, /* tag */ 0);
+        ingest_wbs.writeLogAndData();
+        delegator.addDTFile(file_id, dm_file->getBytesOnDisk(), parent_path);
+
+        WriteBatches wbs(*dm_context->storage_pool, dm_context->getWriteLimiter());
+        auto ref_id = storage_pool->newDataPageIdForDTFile(delegator, __PRETTY_FUNCTION__);
+        wbs.data.putRefPage(ref_id, dm_file->pageId());
+        auto ref_file = DMFile::restore(
+            dm_context->global_context.getFileProvider(),
+            file_id,
+            ref_id,
+            parent_path,
+            DMFileMeta::ReadMode::all());
+        wbs.writeLogAndData();
+        ASSERT_TRUE(segment->ingestDataToDelta(
+            *dm_context,
+            segment->getRowKeyRange(),
+            {ref_file},
+            /* clear_data_in_range */ false));
+
+        ingest_wbs.rollbackWrittenLogAndData();
+    };
+
+    {
+        // let segment's rowkey_range be [60, 90)
+        HandleRange range(60, 90);
+        segment->rowkey_range = RowKeyRange::fromHandleRange(range);
+        // write ColumnFileBig with range [0, 50) and [50, 100)
+        write_column_file_big(0, 100, 2);
```

Comment:
let's generate a segment with [50,150) and ColumnFileBig with 5 packs: [0, 40), [40, 80), [80, 120), [120, 160), [160, 200)

So that we can cover the case that: pack range is out of the range of segment; pack range is overlapped with segment; pack range is covered by segment

Reply:
Addressed